### ISSUE TITLE
VM listing operations fail when node is offline

### ISSUE THREAD

**Issue by xevz** [bpg/terraform-provider-proxmox#692]
**Describe the bug**

If one or more nodes are offline in the cluster, VM listing operations such as the `proxmox_virtual_environment_vms` data source or state refreshing fails as the `/nodes/<node>/` endpoint is used.

**To Reproduce**

1. Set up multi node cluster.
2. Create multiple `proxmox_virtual_environment_vm` resources in the cluster, on multiple nodes
3. Shut off one of the nodes
4. Run configuration with `proxmox_virtual_environment_vms` data source
5. See error
    ```
    Planning failed. Terraform encountered an error while generating this plan.
    
   ╷
   │ Error: error retrieving VMs: received an HTTP 595 response - Reason: No route to host
   │
   │   with module.templates.data.proxmox_virtual_environment_vms.all_vms,
   │   on .terraform/modules/templates/main.tf line 1, in data "proxmox_virtual_environment_vms" "all_vms":
   │    1: data "proxmox_virtual_environment_vms" "all_vms" {}
   │
   ╵
    ```

**Expected behavior**

VMs on the offline node to be listed, but as "shutdown"/"offline" state, and ignored for further operations.

**Additional context**

The GUI seems to use the `/cluster/resources` API endpoint, rather than iterating over  `/nodes/<node>`, which successfully list all VMs in the cluster without having all nodes running.

**Comment by markti** [comment]
I am experiencing this too as this is needed to clone a VM from a packer-built image I think.

**Comment by bpg** [comment]
@markti Hmm, I’m not sure how this use case is related to the cloning you’re trying to do 🤔  
Unless you’re somehow using the `proxmox_virtual_environment_vm` data source in your configuration?

If you have a specific question or are troubleshooting an error, feel free to open a new issue, and I’ll do my best to help!

**Comment by markti** [comment]
Oh yeah I could totally be getting it wrong. Would appreciate any guidance. I've been kicking around this provider on my homelab because it seems much more feature rich than the other one I've been using. I might be doing it wrong, but I am using the VM data source to try and get a reference of the VM ID to the clone nested block. Is there another way?

**Comment by bpg** [comment]
I don’t think you need a data source in your case, the template VM ID can be referenced directly in the clone configuration, like this:
```hcl
...
  clone {
    vm_id = 101
    full  = false
  }
...
```

This works if you already know the template ID beforehand. However, if you’re using an automated process to build template images, the template ID might change with each rebuild (e.g., if a new/random ID is assigned). In such scenarios, you’d need a data source to locate the template VM, for example, by a tag.
